lib/services/alpaca_service.dart

```diff
@@ -7,8 +7,25 @@ class AlpacaService {
   final String _apiSecret = 'ylkwSuQSBuxLfua2Eh0VQaBl0QWtGcdzrB0gprVG';
   final String _baseUrl = 'https://data.alpaca.markets';
 
+  Future<List<String>> searchStock(String query) async {
+    final url = Uri.parse('https://paper-api.alpaca.markets/v2/assets/$query');
+    final response = await http.get(url, headers: {
+      'APCA-API-KEY-ID': _apiKey,
+      'APCA-API-SECRET-KEY': _apiSecret,
+    });
+
+    if (response.statusCode == 200) {
+      final data = jsonDecode(response.body);
+      List<String> results = [];
+      results.add(data['symbol']);
+      return results;
+    } else {
+      throw Exception('Failed to search stock');
+    }
+  }
+
   Future<List<ChartDataPoint>> getChartDataPoints(String symbol) async {
-    final url = Uri.parse('$_baseUrl/v2/stocks/$symbol/bars?timeframe=1D&start=2024-01-01T00%3A00%3A00Z&end=2024-06-24T23%3A59%3A00Z&limit=1000&adjustment=raw&feed=sip&sort=asc');
+    final url = Uri.parse('$_baseUrl/v2/stocks/$symbol/bars?timeframe=30Min&start=2024-06-01T00%3A00%3A00Z&end=2024-12-31T23%3A59%3A00Z&limit=10000&adjustment=raw&feed=iex&sort=asc');
```

Comment:
can put the url into the constants.dart file as a static then dont need to retype